app/src/main/java/a3/com/convo/activities/LoginActivity.java

```diff
@@ -136,47 +163,52 @@ protected void getLikedPageInfo(AccessToken access_token) {
-
-                                if (existingPages.containsKey(id)) {
-                                    // page already exists in Parse, so we just get the object id and add it to their likes array
-                                    user.add("pageLikes", existingPages.get(id));
-                                } else {
-                                    // doesn't exist yet, so we add it to the server
-                                    String category = page.optString("category");
-                                    String name = page.optString("name");
-                                    String coverUrl = page.getJSONObject("cover").optString("source");
-                                    String profUrl = page.getJSONObject("picture").getJSONObject("data").optString("url");
-                                    final Page newPage = Page.newInstance(id, name, profUrl, coverUrl, category);
-
-                                    newPage.saveInBackground(new SaveCallback() {
-
-                                        @Override
-                                        public void done(ParseException e) {
-                                            if (e == null) {
-                                                Log.e("LoginActivity", "Create page success");
-                                                user.add("pageLikes", newPage.getObjectId());
-                                                user.saveInBackground();
+                        final ParseUser user = ParseUser.getCurrentUser();
+                        // initialize empty likes array
+                        user.put(PAGE_LIKES, new ArrayList<String>());
+                        if (json_object != null) {
+                            try {
+                                // convert Json object into Json array
+                                JSONArray likes = json_object.getJSONObject(LIKES).optJSONArray(DATA);
+
+                                for (int i = 0; i < likes.length(); i++) {
+                                    final JSONObject page = likes.optJSONObject(i);
+                                    String id = page.optString(ID);
+
+                                    if (existingPages.containsKey(id)) {
+                                        // page already exists in Parse, so we just get the object id and add it to their likes array
+                                        user.add(PAGE_LIKES, existingPages.get(id));
+                                    } else {
+                                        // doesn't exist yet, so we add it to the server
+                                        String category = page.optString(CATEGORY);
+                                        String name = page.optString(NAME);
+                                        String coverUrl = page.getJSONObject(COVER).optString(SOURCE);
+                                        String profUrl = page.getJSONObject(PICTURE).getJSONObject(DATA).optString(URL);
```

Comment:
What happens if page.getJSONObject(PICTURE) or page.getJSONObject(PICTURE).getJSONObject(DATA) is null?

Reply:
If the profile field is null, we just pass in null to the new page-creating function, where the profileUrl parameter is nullable and we check for null there. Then when we try to load a profile picture, we check to see if it is null before loading.